statsmodels/distributions/empirical_distribution.py

```diff
@@ -144,6 +144,48 @@ def __init__(self, x, side='right'):
         # which would have raised a NameError if hit, so would need to be
         # fixed.  See GH#5701.
 
+class EWCDF(StepFunction):
+    """
+    Return the Empirical Weighted CDF of an array as a step function.
+
+    Parameters
+    ----------
+    x : array_like
+        Observations
+    w : array_like
+        Weights of the observations
```

Comment:
What assumptions are made here? >=0?  Sum >0?  These should be checked.  Should also check that shape is correct.